[NO-JIRA] Add guidelines and time series model (#1186)
SoftwareAG/c8y-docs: content/datahub/working-with-datahub-bundle/aligning-data-modeling-and-offloading.md

{{< /c8y-admon-info >}}

#### Guidelines

When modelling your data, you should generally take the following guidelines into account:

|<div style="width:250px">Description</div>
|:---
|The data type of an attribute should be static as otherwise mixed type constellations may occur.|
|When modelling measurements and large volumes of them are likely to be generated, you should leverage the time series data model. When using this model, the offloading and query performance is typically better compared to the default data model.|
|When modelling measurements you should separate the measurements by specifying measurement types. Then each measurement type can be modelled within a separate offloading pipeline, which in turn leads to a cleaner data architecture in the data lake as well as better query performance.|
|You should avoid offloading lists with many entries as this leads to broad tables in the data lake.|
|You should avoid having a large number of columns in the data lake as this adversely affects query performance and complicates data access in follow-up applications.|

> |When defining attribute names in your data model, you should not use special characters in attribute names. The attribute names are used as column names in the resulting offloading table and special characters may hinder working with those columns in follow-up applications.|

Reviewer: ```suggestion
|When defining attribute names in your data model, avoid special characters in attribute names. The attribute names are used as column names in the resulting offloading table and special characters may hinder working with those columns in follow-up applications.|
```

> |When modelling data within arrays, you must ensure that the position of the values within the array is fix throughout the documents being fed into the platform. Otherwise further processing might run into problems.|

Reviewer: ```suggestion
|When modelling data within arrays, ensure that the position of the values within the array is fix throughout the documents being fed into the platform. Otherwise further processing might run into problems.|
```
